wapitiCore/attack/mod_https_redirect.py

```diff
@@ -0,0 +1,69 @@
+from typing import Optional
+from urllib.parse import urlparse, urlunparse
+
+from httpx import RequestError
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.definitions.https_redirect import NAME, WSTG_CODE
+from wapitiCore.definitions.internal_error import WSTG_CODE as INTERNAL_ERROR_WSTG_CODE
+from wapitiCore.language.vulnerability import Messages
+from wapitiCore.net import Request
+from wapitiCore.net.response import Response
+from wapitiCore.main.log import log_red, log_orange
+
+
+# This module check whether HTTP requests are redirected to HTTPS or not
+class ModuleHttpsRedirect(Attack):
+    """Check for HTTPS redirections."""
+    name = "https_redirect"
+    MSG_VULN_NO_REDIRECT = "No HTTPS redirection"
+    MSG_VULN_REDIRECT = "Redirected to HTTP"
+
+    async def attack(self, request: Request, response: Optional[Response] = None):
+
+        url = urlparse(request.url)
+        # ensure targeting http url
+        http_url = urlunparse(url._replace(scheme='http'))
+        http_request = Request(
+            http_url, request.method,request.get_params,
+            request.post_params, request.file_params, request.encoding,
+            request.enctype, request.referer, request.link_depth
+        )
```

Comment:
Would it work with unpacking ? 

`http_request = Request(http_url, **request)`